sympy/core/evalf.py

```diff
@@ -437,12 +437,8 @@ def calc_part(re_im: 'Expr', nexpr: MPF_TUP):
                     try:
                         as_int(v, strict=False)
                     except ValueError:
-                        try:
-                            [as_int(i, strict=False) for i in v.as_real_imag()]
-                            continue
-                        except (ValueError, AttributeError):
-                            doit = False
-                            break
+                        [as_int(i, strict=False) for i in v.as_real_imag()]
```

Comment:
This changed the flow. If an error raises in this comprehension then `doit` became False, previously. Now the code will stop at the error.

Reply:
Do you know of any examples where this applies?

There are no tests failing so can you conceive of a test case that would be relevant here?

Reply:
thinking on this...

Reply:
Here is an example:
```
In [1]: floor(x).evalf(subs={x:sqrt(2)})
---------------------------------------------------------------------------
ValueError
```
With master:
```
In [1]: floor(x).evalf(subs={x:sqrt(2)})
Out[1]: 1.00000000000000
```

Reply:
I've restored this but made the code a bit clearer.

Reply:
Yes, I was testing similar:
```
options={'subs':{x:pi}}
calc_part(sqrt(x), sqrt(pi).n()._mpf_)
sqrt(x).n(subs={x:pi})
```